### PR TITLE
Update go lib to integrate with addon-framework

### PR DESCRIPTION
**Pull request by qiujian16**
Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

**Review by xiangjingli — APPROVED** [review]

File: pkg/helmrelease/client/client.go [hunk 1]
```diff
@@ -150,11 +149,6 @@ func (c *ownerRefInjectingClient) Build(reader io.Reader, validate bool) (kube.R
 		if useOwnerRef && !containsResourcePolicyKeep(u.GetAnnotations()) {
 			ownerRef := metav1.NewControllerRef(c.owner, c.owner.GroupVersionKind())
 			u.SetOwnerReferences([]metav1.OwnerReference{*ownerRef})
-		} else {
-			err := handler.SetOwnerAnnotations(u, c.owner)
```
Comment (mikeshng) [inline pkg/helmrelease/client/client.go:154]: For the other reviewers. I see this change, and I am ok with it.

File: pkg/utils/kustomize.go [hunk 2]
```diff
@@ -38,7 +38,6 @@ func RunKustomizeBuild(kustomizeDir string) ([]byte, error) {
 
 	options := &krusty.Options{
 		DoLegacyResourceSort: true,
-		UseKyaml:             true,
```
Comment (mikeshng) [inline pkg/utils/kustomize.go:41]: verified with author that this is a deprecated field after lib is upgraded

**Review by mikeshng — APPROVED** [review]
/lgtm

**Pull request merged**